File: scripts/compare_diversity.R
library(vegan)

# Example data: 2 samples x 3 species
# (Matrix is usually sample-rows x species-columns)
# We'll make this numeric matrix in R:
mat <- matrix(c(10, 10, 10,
                5,  0,  5),
              nrow = 2, byrow = TRUE)

# Shannon (default)
vegan_shannon <- diversity(mat, index = "shannon")
vegan_shannon

# Simpson
vegan_simpson <- diversity(mat, index = "simpson")
vegan_simpson

# Inverse Simpson
vegan_inv_simpson <- diversity(mat, index = "invsimpson")
vegan_inv_simpson

File: scripts/compare_diversity.py
import numpy as np
from ecopystats import diversity
import duckdb
import pandas as pd

dune_data = pd.read_csv("ecopystats/data/dune.csv")

print(dune_data)

# Shannon along rows (axis=1 by default)
print(diversity(dune_data, method="shannon"))  
# [1.09861229 0.69314718]

# 'gini-simpson' => 1 - sum(p^2), matches vegan's "simpson"
print(diversity(dune_data, method="gini-simpson"))
# [0.66666667 0.5      ]

# 'simpson' => sum(p^2), but with numbers_equivalent => 1 / sum(p^2) = inverse Simpson
print(diversity(dune_data, method="simpson"))#, numbers_equivalent=True))
# [3. 2.]
